src/Storages/StorageDistributed.cpp

```diff
@@ -1140,22 +1140,20 @@ std::optional<QueryPipeline> StorageDistributed::distributedWriteFromClusterStor
     ContextMutablePtr query_context = Context::createCopy(local_context);
     query_context->increaseDistributedDepth();
 
+    const auto & current_settings = query_context->getSettingsRef();
+    auto timeouts = ConnectionTimeouts::getTCPTimeoutsWithFailover(current_settings);
+
     /// Here we take addresses from destination cluster and assume source table exists on these nodes
-    for (const auto & replicas : getCluster()->getShardsAddresses())
+    for (const auto & replicas : getCluster()->getShardsInfo())
     {
+        /// Skip unavailable hosts if necessary
+        auto try_results = replicas.pool->getMany(timeouts, current_settings, PoolMode::GET_MANY, /*async_callback*/ {}, /*skip_unavailable_endpoints*/ true);
```

Comment:
If the comment on line 1152 is correct. Then `PoolMode::GET_ONE` is sufficient here

Reply:
I guess the comment is either incorrect or misleading (because we get the cluster from `StorageDistributed::getCluster()`), we can ask @nikitamikhaylov

Reply:
Yes, it's copy-paste from https://github.com/ClickHouse/ClickHouse/blob/4315d44517e1ca0412c7dc7c6c39dc1c17881002/src/Storages/StorageReplicatedMergeTree.cpp#L5562-L5565

And looks like we have exactly the same issue in `StorageReplicatedMergeTree::distributedWriteFromClusterStorage` (and that's why we should avoid copy-paste when possible)